.devcontainer/adx/create-adx.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+# Display Help
+Help()
+{
+   echo "Create data explorer cluster for development purpose."
+   echo
+   echo "Syntax: create-adx.sh [-u|h]"
+   echo "options:"
+   echo "u     Unique name used to create resources."
+   echo "h     Print this Help."
+   echo
+}
+
+while getopts u:h flag;
+do
+    case "${flag}" in
+        h) Help
+           exit;;
+        u) unique=${OPTARG};;
+    esac
+done
+
+if [ ! "$unique" ]; then
+    echo "Please add unique name argument."
+    exit;
+fi
+
+# Resource Names
+RESOURCE_GROUP_LOCATION="westeurope"
+RESOURCE_GROUP_NAME="rg-k2-$unique-dev"
+ADX_CLUSTER_NAME="adxk2$unique"
+ADX_DB_NAME="dev"
+STORAGE_ACCOUNT_NAME="stk2$unique"
+STORAGE_CONTAINER_NAME="scripts"
+SERVICE_PRINCIPAL_NAME="sp-k2-$unique"
+
+# Create Resource Group
+RESOURCE_GROUP_SCOPE=$(az group create --location $RESOURCE_GROUP_LOCATION --resource-group $RESOURCE_GROUP_NAME --query id -o tsv)
+
+# Create Data Explorer Cluster
+az extension add -n kusto
+echo "$(tput setaf 2)[1/8] Creating Data Explorer Cluster. Please be patient. Long running operation (10 minutes).$(tput setaf 7)"
+az kusto cluster create \
+    --cluster-name $ADX_CLUSTER_NAME \
+    --sku name="Dev(No SLA)_Standard_D11_v2" tier="Basic" capacity=1 \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --location $RESOURCE_GROUP_LOCATION \
+    --enable-streaming-ingest true \
+    --type="None"
+
+# Create Data Explorer Database
+echo "$(tput setaf 2)[2/8] Creating Data Explorer Database.$(tput setaf 7)"
+az kusto database create \
+    --cluster-name $ADX_CLUSTER_NAME \
+    --database-name $ADX_DB_NAME \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --read-write-database soft-delete-period=P365D hot-cache-period=P31D location=$RESOURCE_GROUP_LOCATION
+
+# Create Service Principal
+echo "$(tput setaf 2)[3/8] Creating Service Principal.$(tput setaf 7)"
+SECRETS=$(az ad sp create-for-rbac --name $SERVICE_PRINCIPAL_NAME --role Contributor --scopes $RESOURCE_GROUP_SCOPE)
+SERVICE_PRINCIPAL_FQN="aadapp=$(echo $SECRETS | jq -r '.appId');$(echo $SECRETS | jq -r '.tenant')"
```

Comment:
So this script also depends on `jq`, you should probably mention it in the doc, or avoid it (using -otsv and cut like in other places, though I will admit it's prone to breaking)

Reply:
Yes, jq has been installed by default on the dev container. But I take your point and will investigate if I can update this code.

Reply:
Fair enough.
So I think that we should still explicitly mention it as a dependency just in case somebody tries that in their own machine (and when I think about it, also the az cli).
And if we're using it we should probably be consistent and change the `-o tsv` we use to jq to be more robust

Reply:
I removed the cut commands but I prefer to keep native way to query azure cli outputs (I mean when only one value is needed).  So, when possible I use : 
--query <property> -o tsv

But, unfortunately, when I cannot use this pattern (for instance $SECRETS where I need multiple values); I switch to jq.